Classes/Hooks/CmdMapLocalizationHook.php

```diff
@@ -0,0 +1,207 @@
+                );
+                if (!empty($translatedContent) && isset($translatedContent['translations'])) {
+                    foreach ($translatedContent['translations'] as $translation) {
+                        if ($translation['text'] != '') {
+                            $value = $translation['text'] ?? $value;
+                            $deepLTranslated = true;
+                            break;
+                        }
+                    }
+                }
+                $translatedElement[$field] = $value ?? $originalElement[$field];
+            }
+            if ($deepLTranslated) {
+                // empty slug field to auto create
+                if ($detectedSlugField !== '') {
+                    $translatedElement[$detectedSlugField] = '';
+                }
+                $data[$table][$translatedElement['uid']] = $translatedElement;
+                $innerDataHandler = GeneralUtility::makeInstance(DataHandler::class);
+                $innerDataHandler->start($data, []);
+                $innerDataHandler->process_datamap();
+                GeneralUtility::makeInstance(PageRepository::class)
+                    ->markPageAsTranslatedWithDeepl($siteInformation['pageUid'], $targetLanguage);
+            }
+        }
+    }
+
+    private function cleanUpWithDefault(
+        string $table,
+        array $originalElement,
+        array $translatedElement,
+        DataHandler $dataHandler,
+        array $tableTCA,
+        array $targetLanguage
+    ): void {
+        // load some defaults
+        $pageId = $table === 'pages' ? $originalElement['uid'] : $originalElement['pid'];
+        $TSConfig = BackendUtility::getPagesTSconfig($pageId)['TCEMAIN.'] ?? [];
+        $tableEntries = $dataHandler->getTableEntries($table, $TSConfig);
+        if (!empty($TSConfig['translateToMessage']) && !($tableEntries['disablePrependAtCopy'] ?? false)) {
+            $translateToMsg = $this->getLanguageService()->sL($TSConfig['translateToMessage']);
+            $translateToMsg = @sprintf($translateToMsg, $targetLanguage['title']);
+        }
+
+        $detectedSlugField = '';
+        foreach ($translatedElement as $field => $value) {
+            if (
+                isset($tableTCA[$field])
+                && isset($tableTCA[$field]['config']['type'])
+                && $tableTCA[$field]['config']['type'] === 'slug'
+            ) {
+                $detectedSlugField = $field;
+                continue;
+            }
+            if (
+                !isset($tableTCA[$field])
+                || !isset($tableTCA[$field]['config']['type'])
+                || (
+                    $tableTCA[$field]['config']['type'] !== 'text'
+                    && $tableTCA[$field]['config']['type'] !== 'input'
+                )
+            ) {
+                continue;
+            }
+            if (!empty($translateToMsg) && !empty($originalElement[$field])) {
```

Comment:
Would replace this block with:

```php
$orignalElementValue = $originalElement[$field] ?? '';
$translatedElement[$field] = $originalElementValue;
if ($translateToMsg !== '' && $orignalElementValue !== '') {
  $translatedElement[$field] = '[' . $translateToMsg . '] ' . $orignalElementValue;
}
```

or

```php
$orignalElementValue = $originalElement[$field] ?? '';
$translatedElement[$field] = ($translateToMsg !== '' && $orignalElementValue !== '')
  ? $translatedElement[$field] = '[' . $translateToMsg . '] ' . $orignalElementValue
  : $orignalElementValue;
```

At least, avoid recurring `null coalesce operator` on the same element/field/data just in a handfull line of code. Better extract and use one variable before. This reduces needed cognitive load and minimized errors lateron for refactorings and further work.

Reply:
Feel free to make a TASK in Core, as this part is even only copied from there and made fit my needs. But your'e right, I changed it.

Reply:
Will adopt it for the core the next days.